diesel/src/mysql/query_builder/query_fragment_impls.rs

```diff
@@ -63,3 +69,71 @@ where
         Ok(())
     }
 }
+
+impl<T> QueryFragment<Mysql, crate::mysql::backend::MysqlOnConflictClause> for DoNothing<T>
+where
+    T: Table,
+    T::PrimaryKey: Column,
+{
+    fn walk_ast<'b>(&'b self, mut out: AstPass<'_, 'b, Mysql>) -> QueryResult<()> {
+        out.push_sql(" UPDATE ");
+        out.push_identifier(<T::PrimaryKey as Column>::NAME)?;
+        out.push_sql(" = ");
+        out.push_identifier(<T::PrimaryKey as Column>::NAME)?;
+        Ok(())
+    }
+}
+
+impl<T, Tab> QueryFragment<Mysql, crate::mysql::backend::MysqlOnConflictClause> for DoUpdate<T, Tab>
+where
+    T: QueryFragment<Mysql>,
+    Tab: Table,
+    Tab::PrimaryKey: Column,
+{
+    fn walk_ast<'b>(&'b self, mut out: AstPass<'_, 'b, Mysql>) -> QueryResult<()> {
+        out.unsafe_to_cache_prepared();
+        out.push_sql(" UPDATE ");
+        if self.changeset.is_noop(out.backend())? {
```

Comment:
Neat! I can't remember why it wasn't something I thought we could do back then: https://github.com/diesel-rs/diesel/blob/e01a9beb2c111d42c46a4343b6444b69db281cda/diesel/src/query_builder/upsert/on_conflict_actions.rs#L83-L85